crates/matrix-sdk/src/event_cache/store.rs

```diff
@@ -352,8 +528,38 @@ mod tests {
             assert_let!(Some((position, event)) = events.next());
             assert_eq!(position.chunk_identifier(), 0);
             assert_eq!(position.index(), 2);
+            assert_eq!(event.event_id().unwrap(), event_id_2);
+
+            assert!(events.next().is_none());
+        }
+
+        // Everything is alright. Now let's insert a duplicated events!
+        room_events.insert_events_at([event_0, event_3], position_of_event_2).unwrap();
+
+        {
+            let mut events = room_events.events();
+
+            // The first `event_id_0` has been removed.
+            assert_let!(Some((position, event)) = events.next());
+            assert_eq!(position.chunk_identifier(), 0);
+            assert_eq!(position.index(), 0);
             assert_eq!(event.event_id().unwrap(), event_id_1);
 
+            assert_let!(Some((position, event)) = events.next());
+            assert_eq!(position.chunk_identifier(), 0);
+            assert_eq!(position.index(), 1);
+            assert_eq!(event.event_id().unwrap(), event_id_0);
```

Comment:
Could we have a small assert helper function (if not a `macro_rules!`) that checks chunk_identifier, index and event id?

Reply:
Sir, yes Sir!

```diff
-            {
-                let mut events = room_events.events();
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 0);
-                assert_eq!(position.index(), 0);
-                assert_eq!(event.event_id().unwrap(), event_id_1);
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 0);
-                assert_eq!(position.index(), 1);
-                assert_eq!(event.event_id().unwrap(), event_id_2);
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 0);
-                assert_eq!(position.index(), 2);
-                assert_eq!(event.event_id().unwrap(), event_id_3);
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 0);
-                assert_eq!(position.index(), 3);
-                assert_eq!(event.event_id().unwrap(), event_id_6);
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 2);
-                assert_eq!(position.index(), 0);
-                assert_eq!(event.event_id().unwrap(), event_id_7);
-
-                assert_let!(Some((position, event)) = events.next());
-                assert_eq!(position.chunk_identifier(), 2);
-                assert_eq!(position.index(), 1);
-                assert_eq!(event.event_id().unwrap(), event_id_8);
-
-                assert!(events.next().is_none());
-            }
+            assert_events_eq!(
+                room_events.events(),
+                [
+                    (event_id_1 at (0, 0)),
+                    (event_id_2 at (0, 1)),
+                    (event_id_3 at (0, 2)),
+                    (event_id_6 at (0, 3)),
+                    (event_id_7 at (2, 0)),
+                    (event_id_8 at (2, 1)),
+                ]
+            );
```

It's pretty useful :-).